mozilla-services/consvc-shepherd
# [SHEP 24] update models and boostr sync script to add cpc cpm rates

## References

[SHEP-24](https://mozilla-hub.atlassian.net/browse/SHEP-24)

## Problem Statement

We store the month and budget for deals , but not the CPC/CPM, and quantity (# of impressions sold). We need this data to calculate and show the Net eCPM value in the Campaigns Overview.

## Proposed Changes

The data is available in the media_plan/line_items boostr endpoint but in order to get the media plan ids we'll have to hit the media_plan endpoint first. Once we have the list of media plan ids, we iterate over all of them and get the monthly line items. We'll pull quantity, rate and rate_type from the line items and update the `boostrdealproduct` object.

The budget column for `boostrdealproduct` has bee updated to `DecimalField` This is because when the total campaign budget is divided into monthly budgets the amount will sometimes have a decimal component representing a dollars and cents value.

## Verification Steps

Run a sync job with `python manage.py sync_boostr_data https://app.boostr.com/api` and check the  `consvc_shepherd_boostrdealproduct` database table for the new quantity , rate and rate_type columns and data.

Note: currently not all deals have media_plans so there will be rows where these values are null. There's also a [known issue](https://mozilla-hub.atlassian.net/browse/SHEP-108) where draft media plans are not synced. We need to decide if it makes sense to sync drafts.

## Visuals

![image](https://github.com/user-attachments/assets/6c7e306a-a6ab-4eda-b918-fcc33dc97769)
